network/src/peer_discovery/primitive.rs

```diff
@@ -0,0 +1,261 @@
+use crate::{primitives::PeerDiscoveryPrimitive, *};
+use async_trait::async_trait;
+use tokio::task::JoinHandle;
+
+struct PeerDiscoveryPrimitiveImpl;
+
+#[async_trait]
+impl PeerDiscoveryPrimitive for PeerDiscoveryPrimitiveImpl {
+    async fn serve(
+        _network_config: &NetworkConfig,
+        _initially_known_peers: Vec<Peer>,
+    ) -> Result<(SharedKnownPeers, JoinHandle<Result<(), Error>>), Error> {
+        unimplemented!();
+    }
+}
+
+#[cfg(test)]
```

Comment:
I think we will have bunch of tests soon. Let's move this to a separate file.

Reply:
done